saleor/plugins/webhook/shipping.py

```diff
@@ -0,0 +1,161 @@
+    defined webhooks by calling a request to each of them one by one.
+    """
+    cached_data = cache.get(cache_key)
+    if cached_data:
+        cached_payload, excluded_shipping_methods = cached_data
+        if payload == cached_payload:
+            return parse_excluded_shipping_methods(excluded_shipping_methods)
+
+    excluded_methods = []
+    # Gather responses from webhooks
+    for webhook in webhooks:
+        response_data = trigger_webhook_sync(
+            event_type,
+            payload,
+            webhook.app,
+            EXCLUDED_SHIPPING_REQUEST_TIMEOUT,
+        )
+        # TODO handle a case when we didn't receive a proper response
+        if response_data:
+            excluded_methods.extend(
+                get_excluded_shipping_methods_from_response(response_data)
+            )
+    cache.set(cache_key, (payload, excluded_methods), CACHE_EXCLUDED_SHIPPING_TIME)
+    return parse_excluded_shipping_methods(excluded_methods)
+
+
+def get_excluded_shipping_data(
+    event_type: str,
+    previous_value: List[ExcludedShippingMethod],
+    payload_fun: Callable[[], str],
+    cache_key: str,
+) -> List[ExcludedShippingMethod]:
+    """Exclude not allowed shipping methods by sync webhook.
+
+    Fetch excluded shipping methods from sync webhooks and return them as a list of
+    excluded shipping methods.
+    The function uses a cache_key to reduce the number of
+    requests which we call to the external APIs. In case when we have the same payload
+    in a cache as we're going to send now, we will skip an additional request and use
+    the response fetched from cache.
+    The function will fetch the payload only in the case that we have any defined
+    webhook.
+    """
+
+    excluded_methods_map: Dict[str, List[ExcludedShippingMethod]] = defaultdict(list)
+    webhooks = _get_webhooks_for_event(event_type)
+    if webhooks:
+        payload = payload_fun()
+
+        excluded_methods_map = get_excluded_shipping_methods_or_fetch(
+            webhooks, event_type, payload, cache_key
+        )
+
+    # Gather responses for previous plugins
+    for method in previous_value:
+        excluded_methods_map[method.id].append(method)
+
+    # Return a list of excluded methods, unique by id
+    excluded_methods = []
+    for method_id, methods in excluded_methods_map.items():
+        reason = None
+        if reasons := [m.reason for m in methods if m.reason]:
+            reason = " ".join(reasons)
+        excluded_methods.append(ExcludedShippingMethod(id=method_id, reason=reason))
+    return excluded_methods
+
+
+def get_excluded_shipping_methods_from_response(
+    response_data: dict,
+) -> List[dict]:
+    excluded_methods = []
+    for method_data in response_data.get("excluded_methods", []):
+        try:
+            raw_id = method_data["id"]
+            typename, _id = graphene.Node.from_global_id(raw_id)
+            if typename == "app":
```

Comment:
Shouldn't be `APP_ID_PREFIX` ?

Reply:
It's the same value. However, in some parts of the code it's referenced to as a string, e.g:
<img width="535" alt="image" src="https://user-images.githubusercontent.com/2566928/153569285-2549e982-14b6-4251-b5a0-1e371385c531.png">

Should we refactor this everywhere? The standard object types are also always represented as strings

Reply:
I would change everywhere to APP_ID_PREFIX. I can imagine we can change this prefix.

Reply:
Agree, we should use `APP_ID_PREFIX` everywhere